mobile/mobile-core/ios-device-discovery.ts

```diff
@@ -0,0 +1,131 @@
+			let deviceIdentifier = iOSDeviceDiscovery.$coreFoundation.convertCFStringToCString(iOSDeviceDiscovery.$mobileDevice.deviceCopyDeviceIdentifier(deviceInfo.dev));
+			iOSDeviceDiscovery.removeDevice(deviceIdentifier);
+		}
+	}
+
+	private static timerCallback(): void {
+		let iOSDeviceDiscovery = $injector.resolve("iOSDeviceDiscovery");
+		iOSDeviceDiscovery.$coreFoundation.runLoopStop(iOSDeviceDiscovery.$coreFoundation.runLoopGetCurrent());
+	}
+
+	private validateResult(result: number, error: string) {
+		if(result !== 0)  {
+			this.$errors.fail(error);
+		}
+	}
+
+	private subscribeForNotifications() {
+		let notifyFunction = ref.alloc(CoreTypes.amDeviceNotificationRef);
+
+		let result = this.$mobileDevice.deviceNotificationSubscribe(this.notificationCallbackPtr, 0, 0, 0, notifyFunction);
+		let error = IOSDeviceDiscovery.APPLE_SERVICE_NOT_STARTED_ERROR_CODE ?
+			"Cannot run and complete operations on iOS devices because Apple Mobile Device Service is not started. Verify that iTunes is installed and running on your system." : "Unable to subscribe for notifications";
+		this.validateResult(result, error);
+		this.$errors.verifyHeap("subscribeForNotifications");
+	}
+
+	private startRunLoopWithTimer(timeout: number): void {
+		let kCFRunLoopDefaultMode = this.$coreFoundation.kCFRunLoopDefaultMode();
+		let timer: NodeBuffer = null;
+
+		if(timeout > 0) {
+			let currentTime = this.$coreFoundation.absoluteTimeGetCurrent() + timeout;
+			timer = this.$coreFoundation.runLoopTimerCreate(null, currentTime , 0, 0, 0, this.timerCallbackPtr, null);
+			this.$coreFoundation.runLoopAddTimer(this.$coreFoundation.runLoopGetCurrent(), timer, kCFRunLoopDefaultMode);
+		}
+
+		this.$coreFoundation.runLoopRun();
+
+		if(timeout > 0) {
+			this.$coreFoundation.runLoopRemoveTimer(this.$coreFoundation.runLoopGetCurrent(), timer, kCFRunLoopDefaultMode);
+		}
+
+		this.$errors.verifyHeap("startRunLoopWithTimer");
+	}
+
+	private createAndAddDevice(devicePointer: NodeBuffer): void {
+		let device = this.$injector.resolve(IOSDevice, {devicePointer: devicePointer});
+		this.addDevice(device);
+	}
+}
+$injector.register("iOSDeviceDiscovery", IOSDeviceDiscovery);
+
+class IOSDeviceDiscoveryStub extends DeviceDiscovery {
+	constructor(private $logger: ILogger,
+		private $staticConfig: Config.IStaticConfig,
+		private $hostInfo: IHostInfo,
+		private error: string) {
+		super();
+	}
+
+	public startLookingForDevices(): IFuture<void> {
+		if(this.error) {
+			this.$logger.warn(this.error);
+		} else if(this.$hostInfo.isLinux) {
+			this.$logger.warn("In this version of the %s command-line interface, you cannot use connected iOS devices.", this.$staticConfig.CLIENT_NAME.toLowerCase());
+		}
+		
+		return Future.fromResult();
+	}
+}
+
+$injector.register("iOSDeviceDiscovery", ($errors: IErrors, $logger: ILogger, $fs: IFileSystem, $injector: IInjector, $iTunesValidator: Mobile.IiTunesValidator, $staticConfig: Config.IStaticConfig, $hostInfo: IHostInfo) => {
```

Comment:
this `register` code must be refactored. Use the "classic" register way:
```TypeScript
$injector.register("iOSDeviceDiscovery", IOSDeviceDiscovery);
```
and move the logic from here directly in the IOSDeviceDiscovery class. Current code will not work very well with Proton setup